### PR TITLE
[1868] Add backlinks and fix page tracker edge case

### PR DESCRIPTION
**Pull request by JonF1**
### Context
* https://trello.com/c/gNixYYfb/1868-s-non-javascript-search-no-back-link

### Changes proposed in this pull request
Add back links to both employing schools and lead schools non-js index views
Change method in page tracker so that if two consecutive edit pages (e.g this school section), then it will goes to previous page instead of last origin page which is what had been happening. This anticipates both the non-js and js journey. 

### Guidance to review
Create both lead and employing schools in your database with FactoryBot
In the Rails console:
`require_relative "config/environment"`
`require "factory_bot_rails"`
`require "faker"`
`FactoryBot.find_definitions rescue nil`
`FactoryBot.create_list(:school, 20, lead_school: true)`
`FactoryBot.create_list(:school, 20)`
Then disable JS

### CODE REVIEW

**Comment by JonF1** [comment]
[[S] Non-javascript search no back link](https://trello.com/c/gNixYYfb/1868-s-non-javascript-search-no-back-link)

**Review by darokel — APPROVED** [review]

**Review by darokel — CHANGES_REQUESTED** [review]
Just checked again and noticed on the employing school page the back link goes back to the review-draft page. Think it should go back to the lead school search

Screen Recording (Google Chrome)

File: app/controllers/trainees/lead_schools_controller.rb [hunk 1]
```diff
@@ -62,6 +65,12 @@ def origin_page_or_next_step
       redirect_url
     end
 
+    def set_origin_page
+      origin_page = session["origin_pages_for_#{trainee.slug}"]
+
+      origin_page << "/trainees/#{trainee.slug}/#{trainee.employing_school.nil? ? 'review-draft' : 'schools/confirm'}".to_s
```
Comment (JonF1) [inline app/controllers/trainees/lead_schools_controller.rb:71]: 😭

**Comment by JonF1** [comment]
> Just checked again and noticed on the employing school page the back link goes back to the review-draft page. Think it should go back to the lead school search

Cheers, should be ready for re-review

**Review by darokel — APPROVED** [review]

**Review by edwardhorsford — APPROVED** [review]

**Pull request merged**